flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/sinks/TableSinkBase.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.api.table.sinks
+
+import org.apache.flink.api.common.typeinfo.TypeInformation
+
+trait TableSinkBase[T] extends TableSink[T] {
```

Comment:
It makes sense to keep TableSink consistent to TableSource. But I think it is a little duplicate as `KafkaTableSink` implement what `TableSinkBase` has implemented again. 

BTW, I think it's better to provide a constructor for KafkaTableSink without fieldNames and fieldTypes, because `Table.writeToSink` will inject this into TableSink automatically.